src/containers/pages/IRS/assignments/index.tsx

```diff
@@ -126,6 +90,27 @@ const IRSAssignmentPlansList = (props: PlanAssignmentsListProps) => {
     tableClass: 'table table-bordered plans-list',
   };
 
+  const asyncRendererProps: AsyncRendererProps<PlanRecord, AsyncGetPlanRecordsOptions> = {
+    data: plans,
+    ifFulfilledRender: () => (
+      <Row>
+        <Col>
+          {plans.length < 1 ? (
+            <span>{NO_PLANS_LOADED_MESSAGE}</span>
+          ) : (
+            <ListView {...listViewProps} />
+          )}
```

Comment:
Is it possible to do this in the `AsyncRenderer` component?

Reply:
This would then make the component re-suable only when we want to show a table. Do you think its a reasonable constraint, coz anyway, I'd be happy to refactor the `AsyncRenderer`. @moshthepitt

Reply:
That is a fair point @p-netm - However how do we make it such that we dont end up rewriting the same code in different components that want to display a list of items?